src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/SchemaGenerator.cs

```diff
@@ -29,11 +38,62 @@ public string GenerateSchema()
+
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(jsonConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(simpleConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(systemdConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(defaultConsoleLoggerOptionsSchema);
+
             string schemaPayload = schema.ToJson();
 
             //Normalize newlines embedded into json
             schemaPayload = schemaPayload.Replace(@"\r\n", @"\n", StringComparison.Ordinal);
             return schemaPayload;
         }
+
+        public static JsonSchema GenerateConsoleLoggerOptionsSchema(JsonSchema consoleFormatterOptions, ConsoleLoggerFormat consoleLoggerFormat)
+        {
+            JsonSchema consoleLoggerOptionsSchema = new JsonSchema();
+
+            JsonSchemaProperty formatterNameProperty = new JsonSchemaProperty();
+            JsonSchemaProperty formatterOptionsProperty = new JsonSchemaProperty();
+
+            JsonSchema formatterOptionsSchema = new JsonSchema();
+            formatterOptionsSchema.Reference = consoleFormatterOptions;
+
+            formatterOptionsProperty.OneOf.Add(formatterOptionsSchema);
+
+            formatterNameProperty.ExtensionData = new Dictionary<string, object>();
+            formatterNameProperty.ExtensionData.Add("const", consoleLoggerFormat.ToString());
+
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterName), formatterNameProperty);
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterOptions), formatterOptionsProperty);
+            consoleLoggerOptionsSchema.RequiredProperties.Add(nameof(ConsoleLoggerOptions.FormatterName));
+
+            return consoleLoggerOptionsSchema;
+        }
+
+        public static JsonSchema GenerateDefaultConsoleLoggerOptionsSchema(JsonSchema consoleFormatterOptions)
+        {
+            JsonSchema consoleLoggerOptionsSchema = new JsonSchema();
+
+            JsonSchemaProperty formatterOptionsProperty = new JsonSchemaProperty();
+            JsonSchemaProperty logToStandardErrorThresholdPropertyDefault = new JsonSchemaProperty();
+
+            JsonSchema formatterOptionsSchema = new JsonSchema();
+            formatterOptionsSchema.Reference = consoleFormatterOptions;
+
+            formatterOptionsProperty.OneOf.Add(formatterOptionsSchema);
+
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.LogToStandardErrorThreshold), logToStandardErrorThresholdPropertyDefault);
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterOptions), formatterOptionsProperty);
+            consoleLoggerOptionsSchema.AllowAdditionalProperties = false;
```

Comment:
This shouldn't be necessary, correct? It's already in the ConsoleLoggerOptions schema.